### PR TITLE
cleanup(tests,vm_test): Use client instead of virtctl

### PR DESCRIPTION
**Pull request by 0xFelix**
### What this PR does

Use the KubeVirt client instead of using virtctl through a NewRepeatableVirtctlCommand where applicable in tests/vm_test.go.

Internally virtctl uses the same client call, so using virtctl does not provide any additional value.

There are more test cases in this file using virtctl which are not changed in this commit because they need to be moved to subresource_api_test.go or should be dropped entirely.

The correctness of API calls made by virtctl start / stop/ restart is already covered in
unit tests specified in `pkg/virtctl/vm/{start,stop,restart}_test.go`.

The goal of this and further PRs is to reduce the amount of unnecessary uses of
virtctl in the e2e tests to stabilize tests, to reduce the number of flakes and to
remove redudant tests.

Before this PR:

NewRepeatableVirtctlCommand is used unnecessarily.

After this PR:

The KubeVirt client is used instead.

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
None
```

### CODE REVIEW

File: tests/vm_test.go [hunk 1]
```diff
@@ -1581,9 +1582,9 @@ status:
 			_, _, err := clientcmd.RunCommand(testsuite.GetTestNamespace(nil), k8sClient, "create", "-f", vmJson)
```
Comment (dankenigsberg) [inline tests/vm_test.go:1582]: It seems that this test covers `kubectl`, and not anything that is kubevirt. Do we need it at all? If we do, can we relegate it to somewhere else? (this file is terribly long as it is).

Comment (0xFelix) [reply tests/vm_test.go:1582]: It is correct that these tests cover `kubectl`. However, before dropping them we need to verify that all of the cases in this context are covered by other tests (or drop them if they don't make any sense). I opened an issue for it: https://github.com/kubevirt/kubevirt/issues/12709

**Review by fossedihelm — APPROVED** [review]

File: tests/vm_test.go [hunk 2]
```diff
@@ -1689,9 +1684,9 @@ status:
 				vmi, err := virtClient.VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name, metav1.GetOptions{})
 				Expect(err).ToNot(HaveOccurred())
 
-				By("Invoking virtctl restart with dry-run option")
-				restartCommand := clientcmd.NewRepeatableVirtctlCommand(virtctl.COMMAND_RESTART, "--namespace", vm.Namespace, "--dry-run", vm.Name)
-				Expect(restartCommand()).To(Succeed())
+				By("Restarting the VM with dry-run option")
```
[the remark this answers is not in the document]
Comment (fossedihelm) [reply tests/vm_test.go:1687]: `Performing dry run restart`?

[the remark this answers is not in the document]
Comment (0xFelix) [reply tests/vm_test.go]: Addressed

File: tests/vm_test.go [hunk 3]
```diff
@@ -1581,9 +1582,9 @@ status:
 			_, _, err := clientcmd.RunCommand(testsuite.GetTestNamespace(nil), k8sClient, "create", "-f", vmJson)
 			Expect(err).ToNot(HaveOccurred())
 
-			By("Invoking virtctl start")
-			startCommand := clientcmd.NewRepeatableVirtctlCommand(virtctl.COMMAND_START, "--namespace", vm.Namespace, vm.Name)
-			Expect(startCommand()).To(Succeed())
+			By("Starting the VM")
```
Comment (xpivarc) [inline tests/vm_test.go:1585]: `should create and delete via command line` (see the `It`) is now miss leading

File: tests/vm_test.go [hunk 4]
```diff
@@ -1614,16 +1615,15 @@ status:
 				_, _, err := clientcmd.RunCommand(testsuite.GetTestNamespace(nil), k8sClient, "create", "-f", vmJson)
 				Expect(err).ToNot(HaveOccurred())
 
-				By("Invoking virtctl start with dry-run option")
-				startCommand := clientcmd.NewRepeatableVirtctlCommand(virtctl.COMMAND_START, "--namespace", vm.Namespace, "--dry-run", vm.Name)
-				Expect(startCommand()).To(Succeed())
+				By("Performing dry run start")
```
Comment (xpivarc) [inline tests/vm_test.go:1618]: `[test_id:7530]in start command` is ambiguous at best. Do we want to rename?

File: tests/vm_test.go [hunk 5]
```diff
@@ -1614,16 +1615,15 @@ status:
...
+				err = virtClient.VirtualMachine(vm.Namespace).Start(context.Background(), vm.Name, &v1.StartOptions{DryRun: []string{metav1.DryRunAll}})
+				Expect(err).ToNot(HaveOccurred())
 
 				_, err = virtClient.VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name, metav1.GetOptions{})
-				Expect(err).To(HaveOccurred())
 				Expect(err).To(MatchError(errors.IsNotFound, "k8serrors.IsNotFound"))
 			})
 
-			DescribeTable("in stop command", func(flags ...string) {
+			DescribeTable("in stop command", func(gracePeriod *int64) {
```
Comment (xpivarc) [inline tests/vm_test.go:1626]: Same as above, the name...

File: tests/vm_test.go [hunk 6]
```diff
@@ -1689,9 +1688,9 @@ status:
 				vmi, err := virtClient.VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name, metav1.GetOptions{})
 				Expect(err).ToNot(HaveOccurred())
 
-				By("Invoking virtctl restart with dry-run option")
-				restartCommand := clientcmd.NewRepeatableVirtctlCommand(virtctl.COMMAND_RESTART, "--namespace", vm.Namespace, "--dry-run", vm.Name)
-				Expect(restartCommand()).To(Succeed())
+				By("Performing dry run restart")
```
Comment (xpivarc) [inline tests/vm_test.go:1691]: Same as above, see the name of `It`

**Review by xpivarc — CHANGES_REQUESTED** [review]
I am ok if we don't want to test `virtctl` in `vm_test.go` but please let's make it clear

File: tests/vm_test.go (hunk 3 above)
Comment (0xFelix) [reply tests/vm_test.go:1585]: This test is still using kubectl, so it is partially command line / partially client. Updated it to drop `via command line`.

File: tests/vm_test.go (hunk 4 above)
Comment (0xFelix) [reply tests/vm_test.go:1618]: Done

File: tests/vm_test.go (hunk 5 above)
Comment (0xFelix) [reply tests/vm_test.go:1626]: Done

**Comment by 0xFelix** [comment]
@xpivarc Addressed

**Pull request merged**